### PR TITLE
Add printSlice function

### PR DESCRIPTION
**Pull request by RuffaloLavoisier**
Like python slice

### CODE REVIEW

File: gfx_2d_print.h [hunk 1]
```diff
@@ -390,6 +390,30 @@ class Graphics2DPrint : public Graphics2D, public Print {
     }
     print(yourNumber);
   }
+  /**
+    @param str String parameter you nedd 'foo' -> E.g char foo[] = "index";
```
Comment (simonmicro) [inline gfx_2d_print.h:394]: Typo

Comment (RuffaloLavoisier) [reply gfx_2d_print.h:394]: Thanks b607c04af63e7d8236d3123df009ce62d1c4821c

File: gfx_2d_print.h [hunk 2]
```diff
@@ -390,6 +390,30 @@ class Graphics2DPrint : public Graphics2D, public Print {
...
+    @param directionHead string cut head
+
+  */
+  void printSlice(char *str, int pos, bool directionHead = false) {
+    uint8_t len = strlen(str);
+
+    if (len > abs(pos)) {
+      if(directionHead){
+        if (pos > 0) 
+          str[pos] = 0;  // 3 : (Hell)o, World -> Hell
```
Comment (RuffaloLavoisier) [inline gfx_2d_print.h:405]: @simonmicro 
How about this is code?

Comment (simonmicro) [reply gfx_2d_print.h:405]: * Maybe you should stay consistent inside the same function body - either using `0` XOR `'\0'`.
* You modify the input `str` variable. IIRC no other `print()` function needs to do that - how about you also try not to do that? This is especially relevant, as you can't use your `printSlice()` function for any `const char*` calls (e.g. from program-memory). Maybe the `String`-class can help you on that :)

**Pull request merged**